dev-support/hbase-personality.sh

```diff
@@ -140,7 +140,15 @@ function personality_modules
 
   clear_personality_queue
 
-  extra="-DHBasePatchProcess"
+  # Set a fork count based off the host cpu count.  Pass maven a -T argument.
+  # Default -T is one thread. 0.5C on an apache box of 16 cores and 2 jenkins
+  # 'executors' per host should make for 8 threads. Setting this here for yetus
+  # to pick up. See
+  # https://yetus.apache.org/documentation/0.11.1/precommit-advanced/#global-definitions
+  # See below for more on -T:
+  # https://cwiki.apache.org/confluence/display/MAVEN/Parallel+builds+in+Maven+3
+  forkcount="0.5C"
+  extra="-T${forkcount}  -Dsurefire.firstPartForkCount=${forkcount} -Dsurefire.secondPartForkCount=${forkcount} -DHBasePatchProcess"
```

Comment:
How does maven `--threads` interact with surefire forks? Isn't it the case that we get `forkCount` maven process thread each building a module simultaneously, and each of those modules is launching up to `forkCount` concurrent test jvms? Means we're looking at 64 concurrent test processes? Seems high to me.

If it is indeed launching like this, I would want `-T{forkCount}` when running `compile/install`, but i want just a single core when running `test`.

I took a read through the [surefire docs](https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html) but I don't have a better handle on this than when I started.

If Maven is instead managing an upper-bound on concurrent test jvms at `${forkcount}`, I say +1.

Reply:
Thanks for the review.

bq. How does maven --threads interact with surefire forks? 

There is no correlation. There may be cases where it acts as a multiplier where modules are independent of each other but generally, there is little opportunity for -T benefit in hbase builds.

I went back and checked a few machines. Couldn't find instance where fork count rose above forkcount limit. Doesn't mean it doesn't happen.

Let me untangle the two.

Reply:
I simplified the patch after the above. Thanks.

Reply:
Hmm... on big box, with 1.0C and no T, build takes 1hr 15mins. With 0.5C and T==2, takes 53minutes. There are probably spurts of lots of processes but interesting that overall time taken goes down significantly. Let me come back and set the T in another follow-on issue.

Reply:
Opened HBASE-24150  Might be more speed up there than here.

Reply:
Good one, I left you a comment over there too.

I wonder if there's a docker param that limits the number of processes a container can launch... could set that to the expected max parallel jvms and see if it breaches.

Reply:
Will poke around....